tests/pytests/unit/test_auth.py

```diff
@@ -0,0 +1,909 @@
+        )
+        # Wrong arg value
+        master_acl_valid_load["arg"] = [
+            "alpha",
+            "gamma",
+            {
+                "kwa": "kwarg #1",
+                "kwb": "kwb",
+                "one_more": "just one more",
+                "__kwarg__": True,
+            },
+        ]
+        await master_acl_clear_funcs.publish(master_acl_valid_load)
+        assert (
+            "error"
+            in master_acl_clear_funcs.event.fire_event.mock_calls.pop(-1).args[0]
+        )
+        assert master_acl_clear_funcs.event.fire_event.mock_calls == []
+        # Wrong kwarg value
+        master_acl_valid_load["arg"] = [
+            "alpha",
+            "beta",
+            "gamma",
+            {
+                "kwa": "kkk",
+                "kwb": "kwb",
+                "one_more": "just one more",
+                "__kwarg__": True,
+            },
+        ]
+        await master_acl_clear_funcs.publish(master_acl_valid_load)
+        assert (
+            "error"
+            in master_acl_clear_funcs.event.fire_event.mock_calls.pop(-1).args[0]
+        )
+        assert master_acl_clear_funcs.event.fire_event.mock_calls == []
+        # Missing arg
+        master_acl_valid_load["arg"] = [
+            "alpha",
+            {
+                "kwa": "kwarg #1",
+                "kwb": "kwb",
+                "one_more": "just one more",
+                "__kwarg__": True,
+            },
+        ]
+        await master_acl_clear_funcs.publish(master_acl_valid_load)
+        assert (
+            "error"
+            in master_acl_clear_funcs.event.fire_event.mock_calls.pop(-1).args[0]
+        )
+        assert master_acl_clear_funcs.event.fire_event.mock_calls == []
+        # Missing kwarg
+        master_acl_valid_load["arg"] = [
+            "alpha",
+            "beta",
+            "gamma",
+            {"kwa": "kwarg #1", "one_more": "just one more", "__kwarg__": True},
+        ]
+        await master_acl_clear_funcs.publish(master_acl_valid_load)
+        assert (
+            "error"
+            in master_acl_clear_funcs.event.fire_event.mock_calls.pop(-1).args[0]
+        )
+        assert master_acl_clear_funcs.event.fire_event.mock_calls == []
+
+
+@pytest.mark.skip_on_windows(reason="PAM eauth not available on Windows")
+async def test_acl_simple_allow(auth_acl_clear_funcs, auth_acl_valid_load):
+    await auth_acl_clear_funcs.publish(auth_acl_valid_load)
+    assert auth_acl_clear_funcs.ckminions.auth_check.call_args[0][0] == [
+        {"alpha_minion": ["test.ping"]}
+    ]
+
+
+async def test_acl_simple_deny(auth_acl_clear_funcs, auth_acl_valid_load):
+    with patch(
+        "salt.auth.LoadAuth.get_auth_list",
+        MagicMock(return_value=[{"beta_minion": ["test.ping"]}]),
+    ):
+        await auth_acl_clear_funcs.publish(auth_acl_valid_load)
+        assert auth_acl_clear_funcs.ckminions.auth_check.call_args[0][0] == [
+            {"beta_minion": ["test.ping"]}
+        ]
```

Comment:
I guess we should just confirm if the yaml in the docstrings can be loaded by pyyaml or not, if it can, leave docstrings as is.